designer/server/src/main/scala/pl/touk/nussknacker/ui/process/processingtype/ScenarioParametersService.scala

```diff
@@ -0,0 +1,119 @@
+package pl.touk.nussknacker.ui.process.processingtype
+
+import cats.data.Validated
+import cats.data.Validated.{invalid, valid}
+import pl.touk.nussknacker.engine.api.component.ProcessingMode
+import pl.touk.nussknacker.engine.api.process.ProcessingType
+import pl.touk.nussknacker.engine.deployment.EngineSetupName
+import pl.touk.nussknacker.engine.util.Implicits.RichTupleList
+import pl.touk.nussknacker.restmodel.scenariodetails.{ScenarioParameters, ScenarioParametersWithEngineSetupErrors}
+import pl.touk.nussknacker.security.Permission
+import pl.touk.nussknacker.ui.security.api.LoggedUser
+import pl.touk.nussknacker.ui.{NotFoundError, NuDesignerError, UnauthorizedError}
+
+class ScenarioParametersService private (
```

Comment:
I don't understand the question :) It is a kind of domain service - it calculates `ScenarioParameters`. What do you mean by "other values"?
- `categoryUnsafe` - Category is the one of these parameters
- `getProcessingTypeWithWritePermission` - This class is also responsible for transition between processing type and scenario parameters in two directions - processingtype -> parameters in factory ('apply') method, opposite direction in this method.

Reply:
I understand the concept behind this class. But if you ask me, what a service is and where it should be located, I would say it is the same as this class except that it can have dependency to external service/repositories etc. and is located in package with other services. I would expect that a class which is located in domain package has name like `ScenarioParameterCalculator` or just `ScenarioParameters`.
Eventually I don't insist on any change, maybe it's only my personal view on this topic.